diff -rupN ../poi-REL_3_15_FINAL/src/java/org/apache/poi/hssf/record/aggregates/DataValidityTable.java ./src/java/org/apache/poi/hssf/record/aggregates/DataValidityTable.java
--- ../poi-REL_3_15_FINAL/src/java/org/apache/poi/hssf/record/aggregates/DataValidityTable.java	2016-10-31 12:29:42.498766498 +0000
+++ ./src/java/org/apache/poi/hssf/record/aggregates/DataValidityTable.java	2016-10-31 12:34:49.712285251 +0000
@@ -65,4 +65,8 @@ public final class DataValidityTable ext
 		_validationList.add(dvRecord);
 		_headerRec.setDVRecNo(_validationList.size());
 	}
+	public void clear() {
+		_validationList.clear();
+		_headerRec.setDVRecNo(0);
+	}
 }
diff -rupN ../poi-REL_3_15_FINAL/src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java ./src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java
--- ../poi-REL_3_15_FINAL/src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java	2016-10-31 12:29:42.494764499 +0000
+++ ./src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java	2016-10-31 12:33:34.034468021 +0000
@@ -128,7 +128,9 @@ public final class HSSFSheet implements
         this._workbook = workbook;
         this._book = workbook.getWorkbook();
     }
-
+    public DataValidityTable getDataValidityTable() {
+        return _sheet.getOrCreateDataValidityTable();
+    }
     /**
      * Creates an HSSFSheet representing the given Sheet object.  Should only be
      * called by HSSFWorkbook when reading in an exisiting file.

